engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/IsValueOfTypeNode.java

```diff
@@ -0,0 +1,96 @@
+package org.enso.interpreter.node.expression.builtin.meta;
+
+import com.oracle.truffle.api.dsl.Fallback;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.interop.UnsupportedMessageException;
+import com.oracle.truffle.api.library.CachedLibrary;
+import com.oracle.truffle.api.nodes.Node;
+import com.oracle.truffle.api.nodes.NodeInfo;
+import com.oracle.truffle.api.profiles.ConditionProfile;
+import org.enso.interpreter.node.expression.builtin.meta.IsSameObjectNode;
+import org.enso.interpreter.node.expression.builtin.meta.TypeOfNode;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.callable.atom.AtomConstructor;
+import org.enso.interpreter.runtime.data.Type;
+import org.enso.interpreter.runtime.library.dispatch.TypesLibrary;
+import org.enso.interpreter.runtime.type.TypesGen;
+
+/** An implementation of the payload check against the expected panic type. */
+@NodeInfo(shortName = "IsValueOfTypeNode")
+public abstract class IsValueOfTypeNode extends Node {
+  private @Child IsSameObjectNode isSameObject = IsSameObjectNode.build();
+  private @Child TypeOfNode typeOfNode = TypeOfNode.build();
+  private final ConditionProfile profile = ConditionProfile.createCountingProfile();
+
+  public static IsValueOfTypeNode build() {
+    return IsValueOfTypeNodeGen.create();
+  }
+
+  public abstract boolean execute(Object panicType, Object payload);
+
+  @Specialization
+  boolean doAtom(AtomConstructor consr, Object payload) {
+    Object panicType = consr.getType();
```

Comment:
Is this correct? You ignore the `AtomConstructor` and take it's type. Would that mean that in
```
t v = case v of
   x : Cons -> "Cons"
   x: Nil -> "Nil"
   _ -> "Anything"
```
we would get `t Cons` to be `"Cons"`, `t Nil` to be also `"Cons"`? Shouldn't this return `"Anything"`?

Reply:
Constructors aren't allowed in type patterns:
```
The name List.Cons resolved to a constructor, but constructors are not allowed in type pattern case.
```